crates/locked-app/src/locked.rs

```diff
@@ -28,6 +68,85 @@ pub struct LockedApp {
     pub components: Vec<LockedComponent>,
 }
 
+fn deserialize_host_requirements<'de, D>(deserializer: D) -> Result<ValuesMap, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    struct HostRequirementsVisitor;
+    impl<'de> serde::de::Visitor<'de> for HostRequirementsVisitor {
+        type Value = ValuesMap;
+
+        fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+            formatter.write_str("struct ValuesMap")
+        }
+
+        fn visit_map<A>(self, mut map: A) -> Result<Self::Value, A::Error>
+        where
+            A: serde::de::MapAccess<'de>,
+        {
+            let mut hr = ValuesMapBuilder::new();
+
+            while let Some(key) = map.next_key::<String>()? {
+                let value: serde_json::Value = map.next_value()?;
+                if value.as_str() == Some(HOST_REQ_OPTIONAL) {
+                    continue;
+                }
+                // TODO: something better with errors, but A::Error is not helpful
+                hr.serializable(key, value).unwrap();
```

Comment:
You could do something like:
```
hr.serializable(key, value).map_err(<A::Error as ::serde::de::Error>::custom)?;
```

Or if you `use serde::de::Error` then:
```
hr.serializable(key, value).map_err(A::Error::custom)?;
```

Reply:
BRIAN

THANK YOU

I had completely not clocked that I needed the trait to do useful things with `A::Error`.  Works like a charm now.